src/app/loadout/plug-drawer/PlugDrawer.tsx

```diff
@@ -5,17 +5,25 @@ import { SearchFilterRef } from 'app/search/SearchBar';
 import { AppIcon, searchIcon } from 'app/shell/icons';
 import { useIsPhonePortrait } from 'app/shell/selectors';
 import { isiOSBrowser } from 'app/utils/browsers';
-import { Comparator } from 'app/utils/comparators';
+import { Comparator, compareBy } from 'app/utils/comparators';
 import _ from 'lodash';
 import React, { RefObject, useCallback, useEffect, useMemo, useRef, useState } from 'react';
 import Sheet from '../../dim-ui/Sheet';
 import '../../item-picker/ItemPicker.scss';
 import Footer from './Footer';
 import PlugSection from './PlugSection';
 
-interface Props {
-  /** A list of plugs the user can choose from */
+export interface PlugsWithMaxSelectable {
+  plugSetHash: number;
   plugs: PluggableInventoryItemDefinition[];
+  maxSelectable: number;
```

Comment:
What does "maxSelectable" mean?

Reply:
Will document it, but basically we associate the max number of selectable mods with each plugset that we build up. We need to do this for the artificer mods, also this makes it much more generic than the previous algorithm which had hardcoded values.

So when we get all the sockets, we group them by the plugset hash. If there are two sockets with the same plugset hash we know we can socket two of those mods. So we place it on this data structure and use it later. So in this case we have one plugset for class item mods and that has two sockets, so we know we can select up to two from that group. Then for the artificer plugset we only have one socket, so we know we can only pick one of them.

The one place this breaks down is slot independent mods. They will always have a count of 5 as there are 5 armour items.